static/js/navigation/main.js

```diff
@@ -39,108 +36,160 @@ if (overlay) {
 }
 
 /**
- * Add event listener to document, to reset dropdown toggle if you click
- * outide the navigation
+ * Add event listener to document, to close all navigation items if user
+ * clicks outside the navigation
  */
-document.addEventListener("click", function (e) {
-  const target = e.target;
+document.addEventListener("click", function (event) {
+  const target = event.target;
   if (target.closest) {
-    if (!target.closest(".p-navigation--sliding, .p-navigation--reduced")) {
-      resetMainToggles(e);
-      navigation.classList.remove("has-menu-open");
+    if (
+      !target.closest(
+        ".p-navigation--sliding, .p-navigation--reduced, .p-navigation.is-secondary"
+      )
+    ) {
+      closeAllNavigationItems();
     }
   }
 });
 
 /**
- * Resets top level toggles back to their original state
- * @param {HTMLElement} excludedToggle - A toggle to ignore
+ * Handles when any toggle is clicked by reseting the other toggles and
+ * managing whether the animation should run
+ * @param {HTMLElement} toggle - The clicked toggle
  */
-function resetMainToggles(excludedToggle) {
-  mainToggles.forEach(function (toggle) {
+function handleToggle(toggle) {
+  const target = document.getElementById(toggle.getAttribute("aria-controls"));
+  if (target) {
+    // check if the toggled dropdown is child of another dropdown
+    const isNested = !!target.parentNode.closest(".p-navigation__dropdown");
```

Comment:
```suggestion
    const isNested = target.parentNode.closest(".p-navigation__dropdown");
```
Can we simplify the check like this? 
If there is no closest ancestor then isNested will be null.
`if (!isNested)` will pass when it is not nested (isNested = null)